e2e-tests/cypress/integration/appeal-submission-appeal-site-address.feature

```diff
@@ -0,0 +1,93 @@
+            | "4 Taylor Road" | ""             | "Bristol"    | "South Glos" | "BS8 1TG" |
+            | "5 Taylor Road" | "Clifton"      | ""           | "South Glos" | "BS8 1TG" |
+
+    Scenario Outline: Prospective Appellant submits a site address without a mandatory field
+        Given the user is prompted for the site address
+        When the user provides their appeal site address as <Address Line 1> and <Address Line 2> and <Town or City> and <County> and <Postcode>
+        Then the user is informed that <reason>
+        #And the user can see that their appeal has NOT been updated with the provided address
+        Examples:
+            | Address Line 1 | Address Line 2 | Town or City | County       | Postcode  | reason                       |
+            | ""             | ""             | ""           | "South Glos" | "BS8 1TG" | "Address Line 1 is required" |
+            | "aaa"          | ""             | ""           | ""           | "BS8 1TG" | "County is required"         |
+            | "aaa"          | ""             | ""           | "South Glos" | ""        | "Postcode is required"       |
+
+    Scenario: We can handle multiple required-field failures at once
+        Given the user is prompted for the site address
+        When the user provides their appeal site address as "" and "" and "" and "" and ""
+        Then the user is informed that "Address Line 1 is required"
+        And the user is informed that "County is required"
+        And the user is informed that "Postcode is required"
+
+    Scenario: We can handle multiple required-field failures for Address Line 1 and County
+        Given the user is prompted for the site address
+        When the user provides their appeal site address as "" and "" and "" and "" and "BS8 1TG"
+        Then the user is informed that "Address Line 1 is required"
+        And the user is informed that "County is required"
+    
+    Scenario: We can handle multiple required-field failures for Address Line 1 and Postcode
+        Given the user is prompted for the site address
+        When the user provides their appeal site address as "" and "" and "" and "South Glos" and ""
+        Then the user is informed that "Address Line 1 is required"
+        And the user is informed that "Postcode is required"
+     
+    Scenario: We can handle multiple required-field failures for County and Postcode
+        Given the user is prompted for the site address
+        When the user provides their appeal site address as "1 Taylor Road" and "" and "" and "" and ""
+        Then the user is informed that "County is required"
+        And the user is informed that "Postcode is required"  
+
+    Scenario Outline: Prospective Appellant provides data that is longer than we accept
+        Given the user is prompted for the site address
+        When the user provides a value which is too long (<component>: <count>)
+        Then the user is informed that <reason>
```

Comment:
I think this step should also make clear that user cannot continue with the address provided as well as the reason for that.

Reply:
updated